cparser/Elab.ml

```diff
@@ -1737,16 +1785,22 @@ let elab_expr vararg loc env a =
       let b1,env = elab env a1 in
       let b2,env = elab env a2 in
       if List.mem AConst (attributes_of_type env b1.etyp) then
-        err "left-hand side of assignment has 'const' type";
+        error "read-only variable is not assignable";
       if not (is_modifiable_lvalue env b1) then
-        err "left-hand side of assignment is not a modifiable l-value";
+        err "expression is not assignable";
       if not (wrap2 valid_assignment loc env b2 b1.etyp) then begin
         if wrap2 valid_cast loc env b2.etyp b1.etyp then
-          warning "assigning a value of type@ %a@ to a lvalue of type@ %a"
-                  Cprint.typ b2.etyp Cprint.typ b1.etyp
+          if wrap2 int_pointer_conversion loc env b2.etyp b1.etyp then
+            warning Int_conversion
+              "incompatible integer to pointer conversion assigning to %a from %a"
+              (print_typ env) b1.etyp (print_typ env) b2.etyp
+          else
+            warning Int_conversion
+              "incompatible pointer to integer conversion assgining to %a from %a"
+               (print_typ env) b1.etyp (print_typ env) b2.etyp
         else
```

Comment:
See my previous comment above.  There are other cases than int->pointer and pointer->int.